src/api.rs

```
@@ -102,6 +102,58 @@ impl<'de> Deserialize<'de> for CheckSquare {
     }
 }
 
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub enum Coordinates {
+    No,
+    Yes,
+}
+
+impl Default for Coordinates {
+    fn default() -> Coordinates {
+        Coordinates::No
+    }
+}
+
+impl<'de> Deserialize<'de> for Coordinates {
+    fn deserialize<D>(deseralizer: D) -> Result<Coordinates, D::Error>
+    where
+        D: de::Deserializer<'de>,
+    {
+        struct CoordinatesVisitor;
+
+        impl<'de> de::Visitor<'de> for CoordinatesVisitor {
+            type Value = Coordinates;
+
+            fn expecting(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
+                fmt.write_str("\"yes\", \"1\", \"no\" or bool")
+            }
+
+            fn visit_str<E>(self, name: &str) -> Result<Coordinates, E>
+            where
+                E: de::Error,
+            {
+                if name == "1" || name == "yes" || name == "true" {
+                    Ok(Coordinates::Yes)
+                } else {
```

Let's report an error if the parameter is not a known falsy name.